dotcom-rendering/src/web/components/SignInGate/displayRule.ts

```diff
@@ -12,7 +12,7 @@ import { CanShowGateProps } from './types';
 // in our case if this is the n-numbered article or higher the user has viewed then set the gate
 export const isNPageOrHigherPageView = (n: number = 2): boolean => {
 	// get daily read article count array from local storage
-	const [dailyCount = {} as DailyArticle] = getDailyArticleCount();
+	const [dailyCount = {} as DailyArticle] = getDailyArticleCount() || [];
```

Comment:
I changed the return type here to `DailyArticleHistory | undefined` because `undefined` is a more deliberate signal that there is _no_ article count. This is consistent with the weekly article count.

Reply:
Just to check thinking here:
`undefined = 0`? A thought on this is that `undefined` is also used to indicate that a reader hasn't given consent? Are we happy saying those are the same things?

Reply:
`getDailyArticleCount` returns `undefined` if there is no article count in local storage, which would be because they haven't consented. On the Supporter Revenue side it's good to be clear about this.
While doing this PR I discovered that the SignInGate (this file) is also using article count - so I've kept the logic the same as before here by defaulting to `[]`.